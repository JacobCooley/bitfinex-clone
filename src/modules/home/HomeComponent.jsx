import React from 'react';
import Button from 'component/Button/Button'
import Input from 'component/Input/Input'
import Loader from 'component/Loader'

class HomeComponent extends React.Component {
	constructor(props) {
		super(props);
		this.state = {
			ticker: 'tBTCUSD'
		}
	}
	
	handleChangeValue = (e) => this.setState({ ticker: e.target.value })
	
	fetchTicker = () => {
		const formattedTicker = this.state.ticker.substr(0, 1).toLowerCase() + this.state.ticker.substr(1, this.state.ticker.length)
		this.props.fetchTickerData(formattedTicker)
	}
	
	render() {
		return (
			<div>
				{this.props.showSpinner ? <Loader/> : null}
				<p>{`Enter the ticker you wish to view`}</p>
				<Input value={this.state.ticker} label={`Tickers:`} name={`tickers`} onChange={this.handleChangeValue}/>
				<Button onClick={this.fetchTicker} text='Get Ticker Data'/>
			</div>
		)
	}
}

export default HomeComponent;