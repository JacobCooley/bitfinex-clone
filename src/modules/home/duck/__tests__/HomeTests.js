import HomeComponent from 'home/HomeComponent'
// import HomeContainer from 'home/HomeContainer'
import React from 'react'
import Adapter from 'enzyme-adapter-react-16'
import { shallow, configure } from 'enzyme'
import configureStore from 'redux-mock-store'
import renderer from 'react-test-renderer'
configure({ adapter: new Adapter() })

jest.mock('component/Tickers/TickersContainer', () => 'tickerscontainer')

describe('<HomeComponent /> --- Snapshot',()=>{
	it('+++ capturing Snapshot of <HomeComponent />', () => {
		const renderedValue =  renderer.create(<HomeComponent />).toJSON()
		expect(renderedValue).toMatchSnapshot();
	});
});

// describe('<HomeComponent /> shallow copy', () => {
// 	let wrapper
//
// 	beforeEach(() => {
// 		wrapper = shallow(<HomeComponent />)
//
// 	})
//
// 	it('+++ render the DUMB component', () => {
// 		expect(wrapper.length).toEqual(1)
// 	})
// })

// describe('<HomeContainer /> shallow copy', () => {
// 	const initialState = {
// 		home: {
// 			count: 0
// 		}
// 	}
// 	const mockStore = configureStore()
// 	let store, container
//
// 	beforeEach(() => {
// 		store = mockStore(initialState)
// 		container = shallow(<HomeContainer store={store}/>)
// 	})
//
// 	it('+++ render the connected(SMART) component', () => {
// 		expect(container.length).toEqual(1)
// 	})
//
// 	it('+++ check Prop matches with initialState', () => {
// 		expect(container.prop('count')).toEqual(initialState.home.count)
// 	})
//
// })
