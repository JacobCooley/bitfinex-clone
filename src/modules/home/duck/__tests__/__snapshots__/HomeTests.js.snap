// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomeComponent /> --- Snapshot +++ capturing Snapshot of <HomeComponent /> 1`] = `
<div>
  <tickerscontainer />
</div>
`;
